### PR TITLE
30179: resetPassword mutation returns generic error

### LINKED ISSUE #30179: [PAP][GraphQL] resetPassword mutation returns generic error
While implementing resetPassword functionality into PWA, an engineer ran into a problem that resulted in this generic error response: `Cannot set the customer's password`. It took digging through the exception log to discover the provided password didn't meet the minimum password requirements, which was not in the error message.

```
<2020-08-11 19:59:26] report.ERROR: Cannot set the customer's password {"exception":"[object] (GraphQL\\Error\\Error(code: 0): Cannot set the customer's password at /app/vendor/webonyx/graphql-php/src/Error/Error.php:174, Magento\\Framework\\GraphQl\\Exception\\GraphQlInputException(code: 0): Cannot set the customer's password at /app/vendor/magento/module-customer-graph-ql/Model/Resolver/ResetPassword.php:121, Magento\\Framework\\Exception\\InputException(code: 0): Minimum of different classes of characters in password is 3. Classes of characters: Lower Case, Upper Case, Digits, Special Characters. at /app/vendor/magento/module-customer/Model/AccountManagement.php:764)"} [>
```

Repro Steps:
1. Execute `requestPasswordResetEmail` mutation and extract token from email
1. Execute `resetPassword` mutation with a weak password in `newPassword` field

Expected: Error is returned with language explaining password didn't meet specific requirements (message from exception log)
 Actual: `Cannot set the customer's password`

image (3) (1)
image (2) (1)

### PR DESCRIPTION
**Pull request by sudheers-kensium**
[GraphQL] resetPassword mutation returns generic error

### Description (*)

While implementing resetPassword functionality into PWA, an engineer ran into a problem that resulted in this generic error response: Cannot set the customer's password. It took digging through the exception log to discover the provided password didn't meet the minimum password requirements, which was not in the error message.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30179: [GraphQL] resetPassword mutation returns generic error

### Manual testing scenarios (*)

1. Execute requestPasswordResetEmail mutation and extract token from email
2. Execute resetPassword mutation with a weak password in newPassword field

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @sudheers-kensium thank you for providing the fix. Could you, please, cover the fix with an API-functional test (or modify an existing one). Basically, the test will check that for different errors, different messages will be returned. 

Thank you!

File: app/code/Magento/CustomerGraphQl/Model/Resolver/ResetPassword.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ public function resolve(
                 $args['newPassword']
             );
         } catch (LocalizedException $e) {
-            throw new GraphQlInputException(__('Cannot set the customer\'s password'), $e);
+            throw new GraphQlInputException(__('The password must be at least 8 characters long, minimum of 3 different classes of characters: Lower Case, Upper Case, Digits, Special Characters.'), $e);
```
Comment (prabhuram93) [inline app/code/Magento/CustomerGraphQl/Model/Resolver/ResetPassword.php]: @sudheers-kensium we try not add new phrases. $e->getMessage() should bubble up the exceptions from the api, which will be more meaningful and is the scope of this bug. $this->customerAccountManagement->resetPassword throws Input Exceptions, you can catch them in the resolver to give meaningful error responses back. For example, I see a bunch of meaningful validations here \Magento\Customer\Model\AccountManagement::checkPasswordStrength
Once you're able to get those messages, you can cover a couple in api functional tests, just to make sure the validation errors are being thrown.

**Review by prabhuram93 — CHANGES_REQUESTED** [review]

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/ResetPasswordTest.php [hunk 2]
```diff
@@ -192,6 +192,52 @@ public function testNewPasswordEmptyValue()
...
+    }
+
+    /**
+     * @magentoApiDataFixture    Magento/Customer/_files/customer.php
+     *
+     * @throws NoSuchEntityException
+     * @throws Exception
+     * @throws LocalizedException
+     */
+    public function testNewPasswordCheckCharactersStrenth()
```
Comment (prabhuram93) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/ResetPasswordTest.php]: minor fix, type here for Strength.

**Review by prabhuram93 — APPROVED** [review]

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8321](https://jira.corp.magento.com/browse/ENGCOM-8321) has been created to process this Pull Request

**Comment by nrkapoor** [comment]
@magento run all tests

**Pull request merged**